src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -78,6 +78,14 @@ private function completeDefinition($id, Definition $definition)
 
             try {
                 if (!$typeHint = $parameter->getClass()) {
+                    // no default value? Then fail
+                    if (!$parameter->isOptional()) {
+                        throw new RuntimeException(sprintf('Unable to autowire argument index %s ($%s) for the service "%s". If this is an object, give it a type-hint. Otherwise, specify this argument\'s value explicitly.', $index, $parameter->name, $id));
```

Comment:
Very minor comment: if `$index` is always a positive integer, should we replace this:

```php
"... argument index %s ($%s) ..."
```

by this?

```php
"... argument index %d ($%s) ..."
```